kDrive/AppRouter.swift

```diff
@@ -596,12 +597,34 @@ public struct AppRouter: AppNavigable {
             fatalError("TODO: lazy load a rootViewController")
         }
 
-        let filePresenter = FilePresenter(viewController: rootViewController)
-        filePresenter.presentPublicShareDirectory(publicShareProxy: publicShareProxy,
-                                                  rootFolder: rootFolder,
-                                                  rootViewController: rootViewController,
-                                                  driveFileManager: driveFileManager,
-                                                  apiFetcher: apiFetcher)
+        guard let rootViewController = window.rootViewController as? MainTabViewController else {
+            fatalError("Root is not a MainTabViewController")
```

Comment:
This will crash if user is not onboarded / is on a "no drive view"

Reply:
Correct, I need to handle the offline mode.

Reply:
This will be done in a dedicated PR, not here. I let you close this comment for now.